src/environment.ts

```diff
@@ -47,6 +47,28 @@ interface SecureEnvironmentOptions {
     distortionMap?: Map<RedProxyTarget, RedProxyTarget>;
 }
 
+// TODO: type this better based on ReflectiveIntrinsicObjectNames
+type ReflectiveIntrinsicsMap = Record<string, any>;
+
+const cachedReflectiveIntrinsicsMap: WeakMap<typeof globalThis, ReflectiveIntrinsicsMap> = WeakMapCreate();
+
+function getReflectiveIntrinsics(global: typeof globalThis): ReflectiveIntrinsicsMap {
+    let reflectiveIntrinsics: ReflectiveIntrinsicsMap | undefined = WeakMapGet(cachedReflectiveIntrinsicsMap, global);
+    if (!isUndefined(reflectiveIntrinsics)) {
+        return reflectiveIntrinsics;
+    }
+    reflectiveIntrinsics = ObjectCreate(null) as ReflectiveIntrinsicsMap;
+    // remapping intrinsics that are realm's agnostic
+    for (let i = 0, len = ReflectiveIntrinsicObjectNames.length; i < len; i += 1) {
+        const name = ReflectiveIntrinsicObjectNames[i];
+        reflectiveIntrinsics[name] = global[name];
+    }
+    return reflectiveIntrinsics;
```

Comment:
Need to set `reflectiveIntrinsics` in the `cachedReflectiveIntrinsicsMap`.

Reply:
good catch, solved